internal/types/path.go

```diff
@@ -48,6 +50,15 @@ func NewPathFromString(s string) Path {
 	return NewPath(path)
 }
 
+// DeriveNewPath returns new Path constructed from given path and path element.
+func DeriveNewPath(path Path, elem string) Path {
```

Comment:
`DeriveNewPath` and `GetNewPath`: what is the difference in the expectation that functions should do?
It's not obvious. Can we use a `GetNewPath` instead?

Reply:
What is `GetNewPath`?

Reply:
> What is `GetNewPath`?

I mean, why did you name this function `DeriveNewPath` not i.e. `GetNewPath`?  
`GetNewPath` is more common.

Reply:
I think we don't need a common name here. `GetNewPath` doesn't add any context. The context here is that we construct a new Path object from the given one and a path element passed to the function. Also `GetNewPath` would be confusing as we already have `NewPath` function.

Reply:
I think that could be `func (p Path) Append(elem string) Path`